### PR TITLE
More lexical declaration support

### PR DESCRIPTION
**Pull request by ariya**

### CODE REVIEW

**Comment by mikesherov** [comment]
:+1: although should probably wait for the unit test split to land.

File: esprima.js [hunk 1]
```diff
@@ -2974,13 +2982,29 @@
 
     // 12.1 Block
 
+    function parseStatementListItem() {
+        if (lookahead.type === Token.Keyword) {
+            switch (lookahead.value) {
+            case 'const':
+            case 'let':
```
Comment (ikarienator) [inline esprima.js]: Note that `let` will soon be an `Identifier`.

Comment (ariya) [inline esprima.js]: I plan to tackle that separately.

**Comment by ariya** [comment]
Rebased against the latest master.

**Comment by ariya** [comment]
@mikesherov Do I need to explicitly sign the CLA (even after the CAA and such)?

**Comment by ikarienator** [comment]
I think we should use some meaningful names for test cases in the future. Otherwise LGTM :+1: !

**Comment by mikesherov** [comment]
@ariya, sign the CLA just so the automated check passes.

**Comment by scottgonzalez** [comment]
I'm curious what the terms of the CAA are and how it affects our checks.

**Pull request closed without merging**